clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -494,14 +498,14 @@ private void updateGroupSubscription(Set<String> topics) {
 
         log.debug("Performing assignment using strategy {} with subscriptions {}", assignor.name(), subscriptions);
 
-        Map<String, Assignment> assignments = assignor.assign(metadata.fetch(), subscriptions);
+        Map<String, Assignment> assignments = assignor.assign(metadata.fetch(), new GroupSubscription(subscriptions)).groupAssignments();
 
         switch (protocol) {
             case EAGER:
                 break;
 
             case COOPERATIVE:
-                adjustAssignment(ownedPartitions, assignments);
+                // TODO need to validate assignment -- make sure no partitions to be revoked were also assigned during this rebalance
```

Comment:
This is something implemented in the (unmerged) [pt 3](https://github.com/apache/kafka/pull/6884/files#diff-e9c1ee46a19a8684d9d8d8a8c77f9005R589) -- left it alone for now